src/uct/api/uct_def.h

```diff
@@ -305,15 +331,111 @@ typedef void
 
 /**
  * @ingroup UCT_RESOURCE
- * @brief Callback to fill the user's private data on the client side.
+ * @brief Callback to process an incoming connection request message on the
+ *        server side listener in a connection manager.
+ *
+ * This callback routine will be invoked on the server side upon receiving an
+ * incoming connection request. It should be set by the server side while
+ * initializing a listener in a connection manager.
+ * This callback has to be thread safe.
+ * Other than communication progress routines, it is allowed to call other UCT
+ * communication routines from this callback.
+ *
+ * @param [in]  listener         Transport listener.
+ * @param [in]  arg              User argument for this callback as defined in
+ *                               @ref uct_listener_params_t::user_data
+ * @param [in]  local_dev_name   Device name which handles the incoming connection
+ *                               request.
+ * @param [in]  conn_request     Connection request handle.
+ * @param [in]  remote_data      Remote data from the client.
+ *
+ */
+typedef void
+(*uct_listener_conn_request_callback_t)(uct_listener_h listener, void *arg,
+                                        const char *local_dev_name,
+                                        uct_conn_request_h conn_request,
+                                        uct_cm_remote_data_t *remote_data);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Callback to process an incoming connection establishment message on the
```

Comment:
Maybe rephrase as "Callback to process an incoming connection request...". I would avoid "message" word everywhere in connection establishment API